### PR TITLE
Heretics cannot be converted, and are immune to cult stun hands. Instead, the cult is rewarded for sacrificing them with the bloody bastard sword, an oversized SPIN2WIN funblade. + Soul Stealing Fantasy Affix

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Heretics can no longer be converted to a cult, as they follow their own Forgotten Gods.
Instead, Nar'Sie will reward the cult for managing to sacrifice one, with the bastard sword.
The bloody bastard sword has been cleaned up codewise and all that. Because it is a free reward instead of a (removed) progression mechanic of cult, it swings just a bit slower during the spin and doesn't have a jaunt. It's still a !fun! swinging sword of hilarity and death.
BLOODY BASTARD https://www.youtube.com/watch?v=ukznXQ3MgN0
Fantasy weapons can now roll "soul-stealing" weapons. They, on killing something, capture its soul inside the item.

- [x] Add fail conditions that instantly end a spin2win, ala how #69592 works

## Why It's Good For The Game

Bloody bastard sword was fun, it made no sense that heretics were valid converts when they're already worshipping a DIFFERENT evil god granting them powers. Should be in a good spot as a nice little antag to antag special interaction. I fucking love antag to antag special interactions, we should have more of 'em

Fantasy affixes are always a neat thing to throw a new component into

## Changelog

:cl:
add: Heretics can no longer be converted to cult. But sacrificing them is very valuable to Nar'Sie, and she will grant special weapons if you manage to do so.
add: Fantasy affixes can also include soul-stealing items!
/:cl:

### CODE REVIEW

File: code/modules/antagonists/cult/hold.txt [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/datum/action/innate/cult/spin2win
```
Comment (tralezab) [inline code/modules/antagonists/cult/hold.txt:1]: i definitely need to remove this file hahaha

**Comment by trexter5555** [comment]
do heretics have any sort of defence against cult instastun hand, or will they just get one clicked round removed?

**Comment by private-tristan** [comment]
> do heretics have any sort of defence against cult instastun hand, or will they just get one clicked round removed?

not as far as I know

**Comment by trexter5555** [comment]
> > do heretics have any sort of defence against cult instastun hand, or will they just get one clicked round removed?
> 
> not as far as I know

heretics should be semi protected from cult hand then, making it just act like a normal heretic hand from the heretic's god protection

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/datums/components/soul_stealer.dm [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/**
+ * ### Soul Stealer component!
+ *
+ * Component that attaches to items, making lethal swings with them steal the victims soul, storing it inside the item.
+ * Used in the cult bastard sword!
+ */
+/datum/component/soul_stealer
+	var/list/souls = list()
```
[the remark this answers is not in the document]
Comment (tralezab) [reply code/datums/components/soul_stealer.dm:8]: I'll weakref it, I like weakrefs

File: code/modules/antagonists/cult/runes.dm [hunk 3]
```diff
@@ -306,8 +306,12 @@ structure_check() searches for nearby cultist structures required for the invoca
 			to_chat(M, span_cultitalic("[sacrificial] is too greatly linked to the world! You need three acolytes!"))
 		log_game("Offer rune with [sacrificial] on it failed - not enough acolytes and target is living or sac target")
 		return FALSE
+	var/datum/antagonist/heretic/heretic_datum
```
[the remark this answers is not in the document]
Comment (tralezab) [reply code/modules/antagonists/cult/runes.dm:309]: interesting... sure, I'll look into it

**Comment by itseasytosee** [comment]
I love this so much. My only wish is more magic v magic PRs in the future and an incentive for heretics to hunt cultist

**Comment by tralezab** [comment]
> > > do heretics have any sort of defence against cult instastun hand, or will they just get one clicked round removed?
> > 
> > 
> > not as far as I know
> 
> heretics should be semi protected from cult hand then, making it just act like a normal heretic hand from the heretic's god protection

> Hard agree with Trexter, all this does is make it so heretics get dunked on by cultist with no ability to defend themselves. That's hardly antag-on-antag interaction, that's just one antag deleting the other from the game.

I wanted to wait for @itseasytosee or another contributor to add the heretic side of this, but sure, I can make them immune to the cult hand to make it a bit harder. I was also interested in the idea of heretics sacrificing cultists bringing a nice bonus like spell recharge and full heal, but that's a bit more boring than what Nar'Sie is doing. Forgotten gods, more like SNOREFEST gods

**Comment by Spookuni** [comment]
Should totally imbue the heretic's soul into the blade and make it a possessed weapon so they can yell/scream/sing as the weapon forged from their body is used for spinning and winning. :P

**Comment by moocowswag** [comment]
Your "why its good for the game" doesnt really say anything about improving player experience, especially for heretic.

**Comment by Shadowflame909** [comment]
Yeah like other people say it should be a soul sword. Narsie does like trapping souls in horrific constructions. So it's not even a stretch!

**Comment by moocowswag** [comment]
If heretics soul is in the sword it opens up the opportunity for heretics to maybe be able to cast certain abilities in favor of who is holding them.

Which doesnt sound op for a joint antag weapon while also preventing bad round removals.

**Comment by tralezab** [comment]
@moocowswag i'm willing to bend on whatever to make it happen, blood cult and clock cult had tons of interactions, but do keep  in mind beforehand it's not like heretics and cults weren't at ends with eachother before this change. This is, at most, incentive. But again I'm taking suggestions here. protection from cult hand? Ok sure why not

>You literally might aswell fucking gib the heretic at round start especially if they are trying to learn.

ok u need to chill out hahahaha

**Comment by CursedBirb** [comment]
> Your "why its good for the game" doesnt really say anything about improving player experience, especially for heretic.
> 
> Heretic shouldnt just be semi protected from cult hand they should be straight up immune, cultists get better melee weapons and armor for near free as a conversion antag while heretic has to fight for every scrap of power that they can get.
> 
> If the issue is heretic being able to too easily sacrifice people as a cultist you have to realize the existence of cult artificially shortens the round to 20-30 minutes (and if they dont end it command staff will with red alert call) meanwhile heretic is designed for a "standard 90 minute round" so heretic has to sacrifice people and progress 3x as fast just when cult is present.
> 
> And now they have to work 3x as fast while a conversion antag that is granted free gear that puts any solo antag to shame's only interaction with them is to round remove them.
> 
> You literally might aswell fucking gib the heretic at round start especially if they are trying to learn.

Why wont we just make a that heretic can sac cultists for one point only?

**Comment by moocowswag** [comment]
> Why wont we just make a that heretic can sac cultists for one point only with sac removing their cult status as heretic is taking their power for themself?

That is an incredibly based idea because it helps compensate for the shorter rounds and I predict 2 common occurrences after this pr. 

1. The heretic sides with crew (which is already a predictable outcome from this pr given that heretic will be a sought-after target for a rare cult item)
2. The heretic stays as a third party and simply scavenges for whatever cultist corpses they can get which is still a big risk/reward since usually a dead cultist means either more cultists are nearby or valid hunting crew or both.

Wait why do i even care about this, its a pr that is adding cult content, maintainers hate cult.

**Comment by itseasytosee** [comment]
I have a PR in the works that gives heretics an example to hunt cultists.

**Comment by tralezab** [comment]
atta fuckin boyyyy

File: code/datums/components/right_click_spin2win.dm [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+	src.spin_cooldown_time = spin_cooldown_time
+	src.on_spin_callback = on_spin_callback
+	src.on_unspin_callback = on_unspin_callback
+	src.start_spin_message = start_spin_message
+	src.end_spin_message = end_spin_message
+
+/datum/component/right_click_spin2win/Destroy(force, silent)
+	QDEL_NULL(on_spin_callback)
+	QDEL_NULL(on_unspin_callback)
+	. = ..()
```
Comment (JohnFulpWillard) [inline code/datums/components/right_click_spin2win.dm:42]:
```suggestion
	return ..()
```

File: code/modules/antagonists/cult/cult_bastard_sword.dm [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+	lefthand_file = 'icons/mob/inhands/64x64_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/64x64_righthand.dmi'
+	inhand_x_dimension = 64
+	inhand_y_dimension = 64
+	actions_types = list()
+	item_flags = SLOWS_WHILE_IN_HAND
+	///are we using our right click ability yet, mr krabs?
+	var/spinning = FALSE
+
+/obj/item/cult_bastard/Initialize()
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/cult/cult_bastard_sword.dm:29]:
```suggestion
/obj/item/cult_bastard/Initialize(mapload)
```

**Review by JohnFulpWillard — CHANGES_REQUESTED** [review]
Moving it to secondary attack means that people won't really know this is a thing unless they fight with rightclick (which i doubt many people do), because we can't expect people to examine every item they get.
At the very least, please add a context menu to the bastard.

**Comment by tralezab** [comment]
As someone who unfortunately played cult when this was in rotation (unfortunate i had to play cult, not use the sword) the most annoying part of the weapon was having to stop fighting, to to the top left of your screen, and click the button. I assure you, this feels far, FAR better.

**Comment by ATH1909** [comment]
might want to make the stun hand immunity only if work if you're a heretic AND are wearing a focus (or something else that lets you cast spells), since I can totally see roundstart cult members getting screwed over hard by happening to choose a non-loud heretic as their first sac.

**Comment by ATH1909** [comment]
> As someone who unfortunately played cult when this was in rotation (unfortunate i had to play cult, not use the sword) the most annoying part of the weapon was having to stop fighting, to to the top left of your screen, and click the button. I assure you, this feels far, FAR better.

click and drag, my man, click and drag

unless that got removed

File: code/modules/antagonists/cult/cult_bastard_sword.dm [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+	icon = 'icons/obj/cult/items_and_weapons.dmi'
+	icon_state = "cultbastard"
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	lefthand_file = 'icons/mob/inhands/64x64_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/64x64_righthand.dmi'
+	inhand_x_dimension = 64
+	inhand_y_dimension = 64
+	actions_types = list()
+	item_flags = SLOWS_WHILE_IN_HAND
+	///are we using our right click ability yet, mr krabs?
```
Comment (ATH1909) [inline code/modules/antagonists/cult/cult_bastard_sword.dm:26]:
```suggestion
	///are ya spinning, son?
```

Comment (Fikou) [reply code/modules/antagonists/cult/cult_bastard_sword.dm:26]: its an auto documentation not a meme place

**Comment by tralezab** [comment]
alllright gonna finish this up

File: code/modules/antagonists/cult/cult_bastard_sword.dm (hunk 6 above)
Comment (tralezab) [reply code/modules/antagonists/cult/cult_bastard_sword.dm:26]: i'm gonna do neither how about that

**Pull request merged**